### PR TITLE
Lower the number of iteration for local_add_mul_fusion

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

**Comment by 0entr0py** [comment]
This PR improves compile time from 4600+s to 700s for the my Conv + Scan [code](https://gist.github.com/0entr0py/48093cb77947b76aebdc69965694c6bd) (see discussion [here](https://groups.google.com/forum/#!topic/lasagne-users/36skUnd1ZOQ)). Will this be merged soon in the master ?

**Comment by nouiz** [comment]
It need to be reviewed by someone else before being merged. I don't expect
problems. It is simple code changes of stuff we already discussed. But the
others are pretty busy, so it could take a few days.

I think it is safe for you to use this branch in the mean time.

File: theano/tensor/opt.py [hunk 1]
```diff
@@ -7140,16 +7140,30 @@ def local_add_mul_fusion(node):
...
+        else:
+            new_inp.append(inp)
+
+    # We ca not compare the number of inputs as Mul and Add could have
+    # 0 or 1 inputs in some corner cases.
+    if fused:
+        output = node.op(*new_inp)
+        copy_stack_trace(node.outputs[0], output)
+
+        # Does the recursion here to help lower the number of
```
Comment (abergeron) [inline theano/tensor/opt.py]: Does -> Do

**Comment by abergeron** [comment]
Ok with small change in the comment to make it not confusing to read.

**Comment by nouiz** [comment]
done

**Pull request merged**